src/lichess.js

```diff
@@ -63,24 +63,53 @@ var makeRequest = (function() {
             var isJSON = request[1];
             var deferred = request[2];
             var promise = null;
-            if (isJSON)  {
-                promise = http.fetchURLIntoJSON(url);
-            } else {
+            try {
                 promise = http.fetchURL(url);
-            }
-            promise.then(function(result) {
-                if (result.response.statusCode === 429) {
-                    requestDelay = 60 * SECONDS;
-                    winston.info("[LICHESS] Last request status was a 429 - we will wait 60 seconds before our next request");
-                }
-                deferred.resolve(result);
-
-                setTimeout(processRequest, requestDelay);
-            }, function(error) {
-                winston.error("[LICHESS] Request failed: " + JSON.stringify(error));
+                promise.then(function(result) {
+                    try {
+                        if (result.response.statusCode === 429) {
+                            requestDelay = 60 * SECONDS;
+                            winston.info("[LICHESS] Last request status was a 429 - we will wait 60 seconds before our next request");
+                        } else {
+                            if (mainQueue.length > 0) {
+                                requestDelay = 2 * SECONDS;
+                            } else {
+                                // Back off a bit if we only have background requests.
+                                requestDelay = 4 * SECONDS;
```

Comment:
I did some napkin calculations:
30 minutes = 1800 seconds / 4 secs = 450. We can process 450 background requests every 30 minutes.  Why 30 minutes?  Because the only background request we have currently is the background rating update and it happens ever 30ish minutes.  we don't have 450 people in the league.  

Also, they don't publish exact load balancer restrictions, but  lichess did change them temporarily to deal with a DDoS over the weekend, and the 4s is closer to what is allowed during that sort of situation.  

If we run into this, I'm going to be in favour of reducing how often we update ratings, not increasing the rate at which we update them.  But up for discussion, definitely.